### PR TITLE
Private/mmeeks/perf simple

### PR DESCRIPTION
**Pull request by mmeeks**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: common/Util-desktop.cpp [hunk 1]
```diff
@@ -424,4 +425,57 @@ void alertAllUsers(const std::string&) {}
...
+    sys = 0;
+#if defined __linux__
+    FILE* cg = fopen("/proc/self/stat", "r");
+    if (cg != nullptr)
+    {
+        char line[4096] = { 0 };
+        if (fgets(line, sizeof(line), cg))
+        {
+            StringVector bits = StringVector::tokenize(line, strlen(line), ' ');
+            cpu = std::atol(bits[13].c_str());
```
Comment (vmiklos) [inline common/Util-desktop.cpp:461]: Aha, man 5 proc explains these are the utime and stime fields, so this measures time, but only for this process, unlike measuring with std::chrono::system_clock::now().

File: common/StringVector.hpp [hunk 2]
```diff
@@ -32,8 +32,9 @@ struct StringToken
 };
 
 /**
- * Safe wrapper around an std::vector<std::string>. Gives you an empty string if you would read past
- * the ends of the vector.
+ * Safe wrapper around an std::vector<std::string>. Gives
```
Comment (vmiklos) [inline common/StringVector.hpp:35]: Nit: not sure this is intended, the header in that file says we wrap at 100 chars, the old one seems to be within that limit.

**Review by vmiklos — APPROVED** [review]
Looks reasonable to me, asking the kernel to measure the process's time for us seems to be a google middle ground between something not time based (e.g. valgrind pseudo cycles) and just measuring system time. Let's wait till CI passes, though.

**Comment by vmiklos** [comment]
https://cpci.cbg.collabora.co.uk:8080/job/github_online_master_debug_vs_co-24.04/1782/console says unit-save-torture failed, retry to see if that's a one-off.

File: test/UnitPerf.cpp [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+#include <thread>
+
+/// Save torture testcase.
+class UnitPerf : public UnitWSD
+{
+    void testPerf();
+
+    void configure(Poco::Util::LayeredConfiguration& config) override
+    {
+//        pfm_initialize();
```
[the remark this answers is not in the document]
Comment (vmiklos) [reply test/UnitPerf.cpp:37]: Ah yes, are the pfm bits something that you first considered, then concluded that /proc/[pid]/stat has enough info for us, or it's rather the other way around, we read /stat as that works already, but would be good to investigate pfm in the future?

**Comment by mmeeks** [comment]
We will need the pfm bits I expect to read the performance counters and get something reasonably accurate; and to do all of this in the same process I think - the /proc stuff is ok-ish; but not terribly helpful I think.

**Pull request merged**